### PR TITLE
react-validated-input - Daniel Yount

### PR DESCRIPTION
**Pull request by TheDanYount**
Recording 2024-08-28 at 17 21 02

### CODE REVIEW

File: react-validated-input/my-app/src/Password.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  const symbolClassNames =
+    errorMessage.length === 0
+      ? 'fa-solid fa-check text-green-500'
+      : 'fa-solid fa-x text-red-500';
+  return (
+    <label htmlFor="password-input" className="font-semibold">
+      Password
+      <br />
+      <input
+        id="password-input"
```
Comment (RobertGardner) [inline react-validated-input/my-app/src/Password.tsx:31]: Why is your cohort so keen to style using the `id`? I've almost never seen that in React code and no prior cohort did this. It must be something Brandon taught you?

File: react-validated-input/my-app/src/Password.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        className="mr-2 p-1 w-[160px] border-[1px] border-black"
+        type="password"
+        value={input}
+        onChange={(e) => setInput(e.target.value)}
+      />
+      <i className={symbolClassNames}></i>
+      {errorMessage.map((str) => (
```
Comment (RobertGardner) [inline react-validated-input/my-app/src/Password.tsx:38]: This is a nice way to get all the error messages to display.

An alternative UI is to just show them one at a time.

**Review by RobertGardner — APPROVED** [review]
Well done. Some suggestions.

File: react-validated-input/my-app/src/Password.tsx (hunk 1 above)
Comment (TheDanYount) [reply react-validated-input/my-app/src/Password.tsx:31]: I don't think I did use id for styling in this exercise. I used it to give the 'for' a target. FreeCodeCamp recommends both couching your inputs inside of labels and connecting them via for & id. As for styling via id in prior exercises, when a single element needs specific styling I guess we could give it a unique class name, I'm not sure why there might be a tendency to use id instead. Brandon didn't tell us to.

**Pull request merged**